next.config.js

```diff
@@ -5,25 +5,33 @@ const { createSecureHeaders } = require('next-secure-headers');
 const nodeExternals = require('webpack-node-externals')
 
 const configs = {
-  webpack(config, { isServer, dev }) {
+  webpack(config, { isServer }) {
     if (isServer) {
       config.externals = [
         nodeExternals({
           allowlist: [/^@xo-union/, /^@babel\/runtime-corejs3\/helpers\/esm\//],
         }),
       ];
     }
-    config.plugins.push(
-      new MiniCssExtractPlugin({
-        filename: 'static/[name].css',
-      })
-    );
+
+    if (!isServer) {
+      config.plugins.push(
+        new MiniCssExtractPlugin({
+          filename: 'static/css/[contenthash].css',
+          chunkFilename: 'static/css/[contenthash].css',
```

Comment:
NOTE: These file names are very important. If you change these, nextjs won't know how to inline link tags during server side rendering (which will cause CSS to load slower, and lead to a flash of unstyled content [FUC])